File: leetcode-04/solutions/cpp/0014.cpp
class Solution {
 public:
  string longestCommonPrefix(vector<string>& strs) {
    if (strs.size() == 0) return "";
    if (strs.size() == 1) return strs[0];

    string ans;
    int min = strs[0].length();
    bool isMatch = true;

    for (int i = 1; i < strs.size(); i++)
      min = std::min(min, (int)strs[i].length());

    for (int i = 0; i < min; i++) {
      char c = strs[0][i];
      for (int j = 1; j < strs.size(); j++)
        if (c != strs[j][i]) {
          isMatch = false;
          break;
        }
      if (!isMatch) break;
      ans += c;
    }

    return ans;
  }
};

File: leetcode-04/solutions/python3/0014.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        if len(strs) == 1:
            return strs[0]

        ans = ""
        min_ = len(strs[0])
        isMatch = True

        for i in range(1, len(strs)):
            min_ = min(min_, len(strs[i]))

        for i in range(min_):
            c = strs[0][i]
            for j in range(1, len(strs)):
                if c != strs[j][i]:
                    isMatch = False
                    break
            if not isMatch:
                break
            ans += c

        return ans
